test/cli/install/bun-yarn.test.ts

```diff
@@ -0,0 +1,236 @@
+  expect(await exited).toBe(0);
+  expect(err).toContain("Saved yarn.lock");
+  expect(out).toContain(`test_pkg_with_prerelease_2@https://github.com/Validark/Validark/raw/master/test_pkg_with_prerelease_2_.tgz`);
+  expect(await getYarnLockContents()).toMatchSnapshot();
+  expect(await getPackageJSONContents()).toMatchSnapshot();
+})
+
+
+it("should work for files with long prerelease versions with no version number in the filename not in a lockfile", async () => {
+	const urls: string[] = [];
+  setHandler(dummyRegistry(urls, { "https://github.com/Validark/Validark/raw/master/test_pkg_with_prerelease_2_.tgz": {} }));
+  const { err, out, exited } = await command("install", "https://github.com/Validark/Validark/raw/master/test_pkg_with_prerelease_2_.tgz", "-y");
+  expect(await exited).toBe(0);
+  expect(err).toContain("Saved yarn.lock");
+  expect(out).toContain(`test_pkg_with_prerelease_2@https://github.com/Validark/Validark/raw/master/test_pkg_with_prerelease_2_.tgz`);
+  expect(await getYarnLockContents()).toMatchSnapshot();
+  expect(await getPackageJSONContents()).toMatchSnapshot();
+})
+
+const yarn_lock_files_from_internet = new Array<[org: string, repo: string]>();
+
+for (const org of await readdir("./resources/yarn")) {
+  for (const repo of await readdir(`./resources/yarn/${org}`)) {
+    yarn_lock_files_from_internet.push([org, repo]);
+  }
+}
+
+it(`should properly parse all ${yarn_lock_files_from_internet.length} yarn.lock files harvested from GitHub`, async () => {
+  const urls: string[] = [];
+  setHandler(dummyRegistry(urls, {}));
+  await makeBasicPackageJSON();
+  for (const [org, repo] of yarn_lock_files_from_internet) {
+    const yarn_lock = await file(`./resources/yarn/${org}/${repo}/yarn.lock`).text();
+    await writeFile(join(package_dir, "yarn.lock"), yarn_lock);
+    const { exited } = await command("install");
+    expect(await exited).toBe(0);
+  }
+})
+
+// Very, very, very slow! It's somewhat like a fuzzer
+/*
+it(`should not segfault for partial yarn lockfiles`, async () => {
+  for (const org of files) {
+    const urls: string[] = [];
+    setHandler(dummyRegistry(urls, {}));
+    await makeBasicPackageJSON();
+    for (const repo of await readdir(`./resources/yarn/${org}`)) {
+      const yarn_lock = await file(`./resources/yarn/${org}/${repo}/yarn.lock`).text();
+      for (let i = 0; ++i <= yarn_lock.length; ) {
+        await writeFile(join(package_dir, "yarn.lock"), yarn_lock.slice(0, i));
+        const { exited } = await command("i");
+        const exit_code = await exited;
+        expect(exit_code === 0 || exit_code === 1).toBe(true);
+      }
+    }
+  }
+})
+*/
+
+const test_yarn_source = `
+"@angular/build-tooling@https://github.com/angular/dev-infra-private-build-tooling-builds.git#414801b1253d48294087fed2bd4d7b07634c2b40", @angular/build-tooling@x:
+  version "0.0.0-7b31625b050e81dcdb575c22f4ed3991922b5bbc"
+  uid "414801b1253d48294087fed2bd4d7b07634c2b40"
```

Comment:
I actually found some `uid` fields in the random yarn.lock files I harvested from GitHub! Didn't find `name` or `registry`, but I did insert code which skips over those fields, if present, in the yarn parser.